tools/log-collector/log_collector.go

```diff
@@ -734,17 +509,92 @@ func (l *LogCollector) CheckIsOpenshift() bool {
 
 // getResourceObjectsWithOwnerRef return all the objects which has ownerRef of CSV
 func (l *LogCollector) getResourceObjectsWithOwnerRef(resourcePath string,
-	resource *apiv1.APIResource) (objects unstructured.UnstructuredList) {
+	resource *apiv1.APIResource) (objects unstructured.UnstructuredList, err error) {
 	allObjects := l.getResourceObjects(resourcePath, resource)
 
 	for _, object := range allObjects.Items {
+
+		if object.GetKind() == SubscriptionKind {
+			startingCSV, _, err := unstructured.NestedString(object.Object, "spec", "startingCSV")
+			if err != nil {
+				log.Errorf("Unable to get startingCSV : %s", err.Error())
+				return objects, err
+			}
+			name, _, nErr := unstructured.NestedString(object.Object, "spec", "name")
+			if nErr != nil {
+				log.Errorf("Unable to get name : %s", nErr.Error())
+				return objects, err
+			}
+
+			if strings.HasPrefix(startingCSV, TrilioPrefix) &&
+				strings.HasPrefix(name, TrilioPrefix) {
+				objects.Items = append(objects.Items, object)
+			}
+		}
+
 		ownerRefs := object.GetOwnerReferences()
 		for idx := range ownerRefs {
-			if strings.HasPrefix(ownerRefs[idx].Name, "k8s-triliovault") &&
-				ownerRefs[idx].Kind == ClusterServiceVersionKind {
+			// Condition Check for CSV as OwnerRef or Subscription as OwnerRef (In Case of InstallPlan)
+			if (strings.HasPrefix(ownerRefs[idx].Name, TrilioPrefix) &&
+				ownerRefs[idx].Kind == ClusterServiceVersionKind &&
+				!excludeResources.Has(object.GetKind())) ||
+				(ownerRefs[idx].Kind == SubscriptionKind &&
+					strings.HasPrefix(ownerRefs[idx].Name, TrilioPrefix) &&
+					object.GetKind() == InstallPlanKind) {
 				objects.Items = append(objects.Items, object)
 			}
 		}
 	}
-	return objects
+	return objects, nil
+}
+
+// checkIfNamespacesExist take all given namespaces from user and checks the same in cluster if it exist
+func (l *LogCollector) checkIfNamespacesExist() (err error) {
+
+	log.Info("Checking Namespaces")
+	set := make(sets.String)
+	var nonExistNs []string
+
+	var namespaces corev1.NamespaceList
+	err = l.k8sClient.List(context.Background(), &namespaces)
+	if err != nil {
+		log.Errorf("%s", err.Error())
+		return err
+	}
+
+	for idx := range namespaces.Items {
+		set.Insert(namespaces.Items[idx].Name)
+	}
```

Comment:
no need of this for loop... use `set := sets.NewString(namespaces...)`

Reply:
resolved